networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/AttestationTopicSubscriber.java

```diff
@@ -19,42 +19,83 @@
 
 import com.google.common.primitives.UnsignedLong;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Iterator;
 import java.util.Map;
 import java.util.Map.Entry;
+import java.util.Set;
+import tech.pegasys.teku.datastructures.validator.SubnetSubscription;
 import tech.pegasys.teku.networking.eth2.Eth2Network;
 import tech.pegasys.teku.util.time.channels.SlotEventsChannel;
 
 public class AttestationTopicSubscriber implements SlotEventsChannel {
-  private final Map<Integer, UnsignedLong> unsubscriptionSlotBySubnetId = new HashMap<>();
+  private final Map<Integer, UnsignedLong> allSubscriptions = new HashMap<>();
+  private final Set<Integer> persistentSubnetIdSet = new HashSet<>();
   private final Eth2Network eth2Network;
 
   public AttestationTopicSubscriber(final Eth2Network eth2Network) {
     this.eth2Network = eth2Network;
   }
 
-  public synchronized void subscribeToCommittee(
+  public synchronized void subscribeToCommitteeForAggregation(
       final int committeeIndex, final UnsignedLong aggregationSlot) {
     final int subnetId = committeeIndexToSubnetId(committeeIndex);
-    final UnsignedLong currentUnsubscriptionSlot =
-        unsubscriptionSlotBySubnetId.getOrDefault(subnetId, ZERO);
+    final UnsignedLong currentUnsubscriptionSlot = allSubscriptions.getOrDefault(subnetId, ZERO);
     if (currentUnsubscriptionSlot.equals(ZERO)) {
       eth2Network.subscribeToAttestationSubnetId(subnetId);
     }
-    unsubscriptionSlotBySubnetId.put(subnetId, max(currentUnsubscriptionSlot, aggregationSlot));
+    allSubscriptions.put(subnetId, max(currentUnsubscriptionSlot, aggregationSlot));
+  }
+
+  public synchronized void subscribeToPersistentSubnets(
+      final Set<SubnetSubscription> newSubscriptions) {
+    boolean shouldUpdateENR = false;
+
+    for (SubnetSubscription subnetSubscription : newSubscriptions) {
+      int subnetId = subnetSubscription.getSubnetId();
+      if (!persistentSubnetIdSet.contains(subnetId)) {
+        persistentSubnetIdSet.add(subnetId);
+        shouldUpdateENR = true;
+      }
```

Comment:
(nit) Another option to make this more succinct / efficient
```suggestion
      shouldUpdateENR = shouldUpdateENR || persistentSubnetIdSet.add(subnetId);
```

Reply:
I think if we did that `persistentSubnetIdSet.add(Subnet)` would not be called if `shouldUpdateENR` is `true`, right? Here we want it to be called regardless of `shouldUpdateENR` being `true` since we want to update `persistenSubnetIdSet`.

Reply:
good point!!  should be:
```
shouldUpdateENR = persistentSubnetIdSet.add(subnetId) || shouldUpdateENR;
```

Reply:
oh hahah. I totally missed that :D

Reply:
Thanks!